### PR TITLE
Workload: Adding view setup job and more debugging tool

### PR DESCRIPTION
**Pull request by samcake**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7739/app_12565_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7739/HighFidelity-Beta-PR12565-7e96608699986b4cd58c98e892a6c208b96ee2e6.exe

**Comment by AndrewMeadows** [comment]
@samcake, the overall changes look fine.  You've introduced some new warnings and we should fix them now rather than later.   Ubuntu is complaining about implicit casts from double.  MacOS and Android are complaining about some template hint subtlety.

File: libraries/task/src/task/Task.h [hunk 1]
```diff
@@ -177,6 +179,10 @@ class Job {
 
     const Varying getInput() const { return _concept->getInput(); }
     const Varying getOutput() const { return _concept->getOutput(); }
+
+    template <class I> void feedInput(const I& in) { _concept->editInput().edit<I>() = in; }
```
Comment (AndrewMeadows) [inline libraries/task/src/task/Task.h]: MacOS and Android complain about this line:
```
09:48:10 /Users/jenkins/Jenkins/workspace/pr-build/label/macos/libraries/task/src/task/Task.h:183:76: error: use 'template' keyword to treat 'edit' as a dependent template name
09:48:10     template <class I> void feedInput(const I& in) { _concept->editInput().edit<I>() = in; }
09:48:10                                                                            ^
09:48:10                                                                            template 
```

File: libraries/workload/src/workload/View.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    View view;
+    view.origin = frustum.getPosition();
+    view.direction = frustum.getDirection();
+    view.setFov(frustum.getFieldOfView());
+
+    return view;
+}
+
+Sphere View::evalRegionSphere(const View& view, float originRadius, float maxDistance) {
+    float radius = (maxDistance + originRadius) / 2.0;
```
Comment (AndrewMeadows) [inline libraries/workload/src/workload/View.cpp]: I spotted this problem in review: implicit cast from `double`.  I was going to let it slide and fix it myself later, but the Ubuntu build is complaining about it.

File: libraries/workload/src/workload/View.h [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    // View attributes:
+
+    // direction
+    glm::vec3 direction{ 0.0f, 0.0f, -1.0f };
+
+    // Max radius
+    float maxRadius{ FLT_MAX };
+
+    // Fov stores the half field of view angle, and tan/cos/sin ready to go, default is fov of 90deg
+    glm::vec4 fov_halfAngle_tan_cos_sin { M_PI_4, 1.0f, M_SQRT2 * 0.5f, M_SQRT2 * 0.5f };
```
Comment (AndrewMeadows) [inline libraries/workload/src/workload/View.h]: Ubuntu sees an implicit cast from `float` to `double` here.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7761/app_12565_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7761/HighFidelity-Beta-PR12565-b6b7cb3394daf9618de38505deb7c7d3d2676eab.dmg

**Pull request merged**